crates/bevy_render/src/render_phase/mod.rs

```diff
@@ -1,14 +1,43 @@
+//! A modular rendering abstraction responsible for queuing, preparing, sorting and drawing entities
+//! as part of separate phases.
+//!
+//! To draw an entity, a corresponding [`PhaseItem`] has to be added to one of the renderers
+//! multiple [`RenderPhase`]s (e.g. opaque, transparent, shadow, etc).
+//! This must be done in the [`RenderStage::Queue`](crate::RenderStage::Queue).
+//! After that the [`RenderPhase`] sorts them in the
+//! [`RenderStage::PhaseSort`](crate::RenderStage::PhaseSort).
+//! Finally the [`PhaseItem`]s are rendered using a single [`TrackedRenderPass`], during the
+//! [`RenderStage::Render`](crate::RenderStage::Render).
+//!
+//! Therefore each [`PhaseItem`] is assigned a [`Draw`] function.
+//! These set up the state of the [`TrackedRenderPass`] (i.e. select the
+//! [`RenderPipeline`](crate::render_resource::RenderPipeline), configure the
+//! [`BindGroup`](crate::render_resource::BindGroup)s, etc.) and then issue a draw call,
+//! for the corresponding [`PhaseItem`].
+//!
+//! The [`Draw`] function trait can either be implemented directly or such a function can be
+//! created by composing multiple [`RenderCommand`]s.
+
 mod draw;
 mod draw_state;
+mod rangefinder;
 
-use bevy_ecs::entity::Entity;
 pub use draw::*;
 pub use draw_state::*;
+pub use rangefinder::*;
 
-use bevy_ecs::prelude::{Component, Query};
-use bevy_ecs::world::World;
+use bevy_ecs::prelude::*;
+use std::ops::Range;
 
-/// A resource to collect and sort draw requests for specific [`PhaseItems`](PhaseItem).
+/// A render phase sorts and renders all [`PhaseItem`]s (entities) that are assigned to it.
```

Comment:
```suggestion
/// A collection of all rendering commands for a single rendering phase for a single view.
///
/// These commands are included as individual [`PhaseItem`] that encodes how to draw one ore more 
/// entities to the screen.
```

Reply:
It is not really collecting rendering commands, but rather `PhaseItems`.
How about 
```rust
/// A collection of all [`PhaseItem`]s for a single rendering phase for a single view.
```

Reply:
In a way, each phase item *is* a bundle of rendering commands. As a summary of what the type does, I'd rather us avoid referential descriptions, especially when it's not clear to the normal user what a `PhaseItem` is from a glance.

Reply:
I use the term `rendering instructions` instead of `rendering commands` because the latter are stored in `DrawFunctionsInternal` instead. I think this is a good compromise. Do you agree?

```
/// A collection of all rendering instructions, that will be executed by the GPU, for a
/// single rendering phase for a single view.
```

Reply:
Works for me, though it could be a bit confusing since "instructions" usually imply CPU/GPU machine instructions.

Reply:
Mhh, yeah both are not ideal. I will change it back to commands if other reviewers agree that this is the better term. 👍